### DISCUSSION TITLE
7182 - Dúvida no OrderHandlerTests

### DISCUSSION THREAD

**Discussion by thiagocajadev** [balta-io/forum#281] [Tira Dúvidas] [answered]
Olá Balta, estou fazendo debug nos casos de testes. No último teste, o handler sempre retorna verdadeiro, mesmo que o customer não atenda os requisitos de tamanho de caracteres:

```csharp
// Exemplo no repo do github
  [TestMethod]
        [TestCategory("Handlers")]
        public void Dado_um_comando_valido_o_pedido_deve_ser_gerado()
        {
            var command = new CreateOrderCommand();
            command.Customer = "12345678"; // O tamanho deveria ser 11 correto?
            command.ZipCode = "13411080";
            command.PromoCode = "12345678";
            command.Items.Add(new CreateOrderItemCommand(Guid.NewGuid(), 1));
            command.Items.Add(new CreateOrderItemCommand(Guid.NewGuid(), 1));

            var handler = new OrderHandler(
                _customerRepository,
                _deliveryFeeRepository,
                _discountRepository,
                _productRepository,
                _orderRepository);

            handler.Handle(command);
            Assert.AreEqual(handler.Valid, true);
        }
```

Alterei o Assert para false, para um novo debug. O fluxo de execução exibe a notificação de erro no comando, porém o handler continua válido:

![OrderHandlerTest](https://user-images.githubusercontent.com/51033018/224532458-70bbdf65-75b8-4fa0-ba58-df3e6e8f91f8.png)

Queria entender se esse comportamento é esperado ou se é preciso adicionar verificações extras no handler para capturar as notificações genéricas.

[7182](https://github.com/balta-io/7182/tree/master)/[good](https://github.com/balta-io/7182/tree/master/good)/[Store.Tests](https://github.com/balta-io/7182/tree/master/good/Store.Tests)/[Handlers](https://github.com/balta-io/7182/tree/master/good/Store.Tests/Handlers/OrderHandlerTests.cs)

Desde já agradeço, estou aprendendo muito com os conteúdos da plataforma .🥇

**Comment by andrebaltieri** [comment, marked as answer]
@thiagokj boa noite! Precisa capturar as exceções genéricas também. Não sei qual erro era para ter dado, pois o código das notificações está fora dos testes, mas de qualquer forma, se há um risco de ocorrer uma exception tratada, você pode testar ela sim.
